webpack.config.js

```diff
@@ -1,27 +1,59 @@
 const webpack = require('webpack');
-const {join} = require('path');
+
+class FallbackRequireWebpackPlugin {
+  apply(compiler) {
+	compiler.hooks.compilation.tap('FallbackRequireWebpackPlugin', (compilation) => {
+		const {renderRequire} = webpack.javascript.JavascriptModulesPlugin.getCompilationHooks(compilation);
+		renderRequire.tap(
+			'FallbackRequireWebpackPlugin',
+			(content, renderContext) => content.replace(
+				/\s*return module\.exports;\s*/,
+				'\nreturn (module.exports && module.exports.name === "webpackEmptyContext") ? require(moduleId) : module.exports;',
+			),
+		);
+	});
+  }
+}
 
 module.exports = {
+	mode: 'production',
 	target: 'node',
 	context: __dirname,
-	entry: './src/plop.js',
+	entry: {
+		main: {
+			import: './src/plop.js',
+			filename: './dist/plop.js',
+		},
+		bin: {
+			dependOn: 'main',
+			import: './src/bin/plop.js',
+			filename: './bin/plop.js',
+		},
+	},
```

Comment:
Double entry point power!

The bin utilizes the install chunks, require definition, and exports from the main entry point

Reply:
Love it - makes sense to me!